src/nvim/lua/executor.c

```diff
@@ -1641,19 +1641,19 @@ bool nlua_is_deferred_safe(void)
   return in_fast_callback == 0;
 }
 
-/// Run lua string
+/// Executes Lua code.
 ///
-/// Used for :lua.
+/// Implements `:lua` and `:lua ={expr}`.
 ///
-/// @param  eap  Vimscript command being run.
+/// @param  eap  Vimscript `:lua {code}`, `:{range}lua`, or `:lua ={expr}` command.
 void ex_lua(exarg_T *const eap)
   FUNC_ATTR_NONNULL_ALL
 {
   if (eap->addr_count > 0 || *eap->arg == NUL) {
     if (eap->addr_count > 0 && *eap->arg == NUL) {
       cmd_source_buffer(eap, true);
     } else {
-      semsg(_(e_invarg2), "exactly one of {chunk} and {range} required");
+      semsg(_(e_invarg2), "exactly one of {chunk} or {range} is required");
```

Comment:
Don't care either way but I deliberately left "is" out because most if not all "required" error messages don't include it.

Reply:
Makes sense. `git grep -P 'emsg\b.*\brequired\b'` shows that other cases use the [simple language](https://github.com/neovim/neovim/pull/27167#discussion_r1466670006). But with the "exactly one of" explanation, omitting "is" sounds pretty weird.

Reply:
"either ... or" if you want to stress the xor?

(otherwise I don't see an issue with the missing "is"; sounds fine to me)

Reply:
> shows that other cases use the simple language

Yes but I don't think that's explicit enough in telling what's wrong when both `{chunk}` and `{range}` are supplied. I think we either need a separate error message in that case or keep the "exactly one of" (or alternative) phrase. If "is" is required to make that phrase more clear then so be it.

Reply:
removed "is" for now.